CMakeLists.txt

```diff
@@ -227,61 +231,17 @@ if(ENABLE_QT)
     set(ENABLE_QT_COM_INTEROP OFF)
     set(ENABLE_QT_DBUS_INTEROP OFF)
 
-    if(USE_QT5)
-        set(QT5_REQUIRED_MODULES Core Gui Widgets Network LinguistTools)
-        set(QT5_OPTIONAL_MODULES DBus AxContainer AxServer)
-        foreach(M ${QT5_REQUIRED_MODULES})
-            find_package(Qt5${M} QUIET)
-            if(Qt5${M}_FOUND)
-                if(NOT M STREQUAL "LinguistTools")
-                    list(APPEND QT_TARGETS Qt5::${M})
-                endif()
-            else()
-                set(QT_TARGETS)
-                break()
-            endif()
-        endforeach()
-        if(QT_TARGETS)
-            foreach(M ${QT5_OPTIONAL_MODULES})
-                find_package(Qt5${M} QUIET)
-                if(Qt5${M}_FOUND)
-                    list(APPEND QT_TARGETS Qt5::${M})
-                endif()
-            endforeach()
-            if(Qt5AxContainer_FOUND AND Qt5AxServer_FOUND)
-                set(ENABLE_QT_COM_INTEROP ON)
-            endif()
-            if(Qt5DBus_FOUND)
-                set(ENABLE_QT_DBUS_INTEROP ON)
-            endif()
-        endif()
-    else()
-        set(QT4_REQUIRED_MODULES QtCore QtGui QtNetwork)
-        set(QT4_OPTIONAL_MODULES QtDBus QAxContainer QAxServer)
-        find_package(Qt4 4.6.2 QUIET COMPONENTS ${QT4_REQUIRED_MODULES} OPTIONAL_COMPONENTS ${QT4_OPTIONAL_MODULES})
-        foreach(M ${QT4_REQUIRED_MODULES})
-            string(TOUPPER "${M}" M_UPPER)
-            if(QT_${M_UPPER}_FOUND)
-                list(APPEND QT_TARGETS Qt4::${M})
-            else()
-                set(QT_TARGETS)
-                break()
-            endif()
-        endforeach()
-        if (QT_TARGETS)
-            foreach(M ${QT4_OPTIONAL_MODULES})
-                string(TOUPPER "${M}" M_UPPER)
-                if(QT_${M_UPPER}_FOUND)
-                    list(APPEND QT_TARGETS Qt4::${M})
-                endif()
-            endforeach()
-            if(QT_QAXCONTAINER_FOUND AND QT_QAXSERVER_FOUND)
-                set(ENABLE_QT_COM_INTEROP ON)
-            endif()
-            if(QT_QTDBUS_FOUND)
-                set(ENABLE_QT_DBUS_INTEROP ON)
-            endif()
-        endif()
+    find_package(Qt5 REQUIRED COMPONENTS Core Gui Widgets Network LinguistTools)
```

Comment:
As I mentioned before, `REQUIRED` here breaks `cmake ... -DENABLE_QT=AUTO` case (which is the default). If `AUTO` is used and Qt wasn't found, no configuration error should be generated, result should be that Qt client just isn't built.